zomes/chat/src/batching_helper.rs

```diff
@@ -42,44 +42,54 @@ pub fn get_message_links(
     let mut depth = 0;
     while links.len() < target_count && current_search_path.as_ref().len() >= root_path_length {
         if current_search_path.exists()? {
-            // let earliest_seen_child_segment =
-            //     last_segment_from_path(&earliest_seen_child_path).unwrap();
+            let earliest_seen_child_segment =
+                last_segment_from_path(&earliest_seen_child_path).unwrap();
             let children = current_search_path.children()?;
-            // This was just an optimization which we are going to loose
-            // children.retain(|child_link| {
-            //     link_is_earlier(child_link, earliest_seen_child_segment).unwrap_or(false)
-            // });
+            debug!(
+                "current_search_path {:?} depth {} children {:?}",
+                current_search_path,
+                depth,
+                children
+                    .iter()
+                    .map(|l| (&l.tag, l.timestamp))
+                    .collect::<Vec<_>>()
+            );
+            let mut children = children
+                .into_iter()
+                .filter_map(|l| path_component_from_link(&l).ok().map(|c| (c, l))) // filter out non-path links
```

Comment:
Once you get to the channel level where each child represents a year, children contains both `ChannelInfo` links and path links. We only want the path links, so I'm filtering them here with `filter_map`